.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,93 @@
+name: Pre-commit actions for Delphix Virtualization Platform SDK
+
+on: [pull_request]
+
+jobs:
+  pytest:
+    name: Test ${{ matrix.package }} on ${{ matrix.os }} using pytest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [2.7]
+        os: [ubuntu-latest, macos-latest]
+        package: [common, libs, platform, tools]
+
+    steps:
+    - name: Checkout ${{ matrix.package }} project
+      uses: actions/checkout@v1
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install ${{ matrix.package }} dependencies
+      working-directory: ${{ matrix.package }}
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt --index-url https://pypi.org/simple --extra-index-url https://test.pypi.org/simple
+
+    - name: Install ${{ matrix.package }} project in editable mode
+      working-directory: ${{ matrix.package }}
+      run: |
+        pip install -e . --index-url https://pypi.org/simple --extra-index-url https://test.pypi.org/simple
+
+    - name: Test ${{ matrix.package }} project with pytest
+      working-directory: ${{ matrix.package }}
+      run: |
+        pip install pytest
+        python -m pytest src/test/python
+
+  lint:
+    name: Lint ${{ matrix.package }}
+
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        package: [tools]
+        
+    steps:
+      - name: Checkout ${{ matrix.package }}
+        uses: actions/checkout@v1
+    
+      - name: Set up Python 2.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 2.7
+   
+      - name: Install flake8
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+        
+      - name: Run flake8 on src directory
+        working-directory: ${{ matrix.package }}
+        run: python -m flake8 src/main/python --max-line-length 88
+    
+      - name: Run flake8 on test directory
+        working-directory: ${{ matrix.package }}
+        run: python -m flake8 test/main/python --max-line-length 88
+
+  #format:
```

Comment:
How much work would it be to run `yapf` which is what we use now until we check in the formatting script? If it is much work, we can hold off since the format script should be coming soon.

Reply:
There are lot of files ( more than 10 ) that are failing with formatting errors. We can take care of this with format script.